### PR TITLE
chore: update map and coding component README with pure ESM approach

### PR DESCRIPTION
**Pull request by manugadde**
Adopt the standard ESM pattern of JS SDK for component samples.

The following changes have been made:
- Updated map and coding component vite, vue, webpack and react samples to individually load components.
- Updated  the respective README files to include:
     - a note on recommended pattern
     - registering the components using:`Loading All Components` approach with sample code.
     - link to https://developers.arcgis.com/javascript/latest/get-started-npm/#components

### CODE REVIEW

File: component-samples/coding-components/samples/react/README.md [hunk 1]
```diff
@@ -10,4 +10,24 @@ The project was created using [`yarn create vite`](https://vitejs.dev/guide/#sca
...
 Run `npm install` and then start adding modules.
 
-For a list of all available `npm` commands see `scripts` in `package.json`, e.g. `npm run build`.
\ No newline at end of file
+For a list of all available `npm` commands see `scripts` in `package.json`, e.g. `npm run build`.
+
+***Note:*** The sample demonstrates the recommended pattern for loading ArcGIS Map Components by individually loading each component using the pure ESM approach.
+
+### Loading All Components
+To register all the components once, use the following approach:
```
Comment (omarkawach) [inline component-samples/coding-components/samples/react/README.md:18]:
```suggestion
The JavaScript Maps SDK also offers a convenience pattern useful for quick testing and prototyping. You can register all components at once using the following approach:
```

File: component-samples/coding-components/samples/react/README.md [hunk 2]
```diff
@@ -10,4 +10,24 @@ The project was created using [`yarn create vite`](https://vitejs.dev/guide/#sca
...
+
+ import { defineCustomElements as defineCalciteElements } from "@esri/calcite-components/dist/loader";
+ import { defineCustomElements as defineCodingElements } from "@arcgis/coding-components/dist/loader";
+
+//define custom elements in the browser, and load the assets from the CDN
+
+ defineCalciteElements(window, { resourcesUrl: "https://js.arcgis.com/calcite-components/2.8.5/assets" });
+ defineCodingElements(window, { resourcesUrl: "https://js.arcgis.com/coding-components/4.30/assets" });
+```
+Recommended to use this ES modules (ESM) via the ArcGIS CDN way of importing only for testing and prototyping.
```
Comment (omarkawach) [inline component-samples/coding-components/samples/react/README.md:31]:
```suggestion
```

File: component-samples/coding-components/samples/react/README.md [hunk 3]
```diff
@@ -10,4 +10,24 @@ The project was created using [`yarn create vite`](https://vitejs.dev/guide/#sca
...
+
+For more details on registering components, please refer to the [ArcGIS for JavaScript API documentation](https://developers.arcgis.com/javascript/latest/get-started-npm/#components).
```
Comment (omarkawach) [inline component-samples/coding-components/samples/react/README.md:33]:
```suggestion
For more details on using the SDK, please refer to the [ArcGIS Maps SDK for JavaScript documentation](https://developers.arcgis.com/javascript/latest/get-started-overview/).
```

File: component-samples/coding-components/samples/react/src/index.tsx [hunk 4]
```diff
@@ -17,12 +17,16 @@ import React from "react";
...
-defineCalciteElements(window, { resourcesUrl: "https://js.arcgis.com/calcite-components/2.8.5/assets" });
+// Individual imports for each component
+import { setArcgisAssetPath as setCodingComponentsAssetPath } from "@arcgis/coding-components/dist/components";
+import "@arcgis/coding-components/dist/components/arcgis-arcade-editor";
+
+import { setAssetPath as setCalciteComponentsAssetPath } from '@esri/calcite-components/dist/components';
+import "@esri/calcite-components/dist/components/calcite-scrim";
+
+setCodingComponentsAssetPath("https://js.arcgis.com/coding-components/4.30/assets");
+setCalciteComponentsAssetPath("https://js.arcgis.com/calcite-components/2.8.0/assets");
```
Comment (omarkawach) [inline component-samples/coding-components/samples/react/src/index.tsx:29]:
```suggestion
// Individual imports for each component
import "@arcgis/coding-components/dist/components/arcgis-arcade-editor";
import "@esri/calcite-components/dist/components/calcite-scrim";

import { ArcgisArcadeEditor } from "@arcgis/coding-components-react";
import { CalciteScrim } from "@esri/calcite-components-react";
```

You'll have to update `react/src/components/ArcadeEditor.tsx`

You don't need to set an asset path when individually importing components, we pick up CDN hosted assets by default.

File: component-samples/coding-components/samples/react/README.md [hunk 5]
```diff
@@ -10,4 +10,24 @@ The project was created using [`yarn create vite`](https://vitejs.dev/guide/#sca
...
\ No newline at end of file
+For a list of all available `npm` commands see `scripts` in `package.json`, e.g. `npm run build`.
+
+***Note:*** The sample demonstrates the recommended pattern for loading ArcGIS Map Components by individually loading each component using the pure ESM approach.
+
+### Loading All Components
+To register all the components once, use the following approach:
+
+```
+// Replace the imports in main.js
```
Comment (omarkawach) [inline component-samples/coding-components/samples/react/README.md:21]:
```suggestion
// Replace the individual imports with defineCustomElements()
```

**Review by omarkawach — CHANGES_REQUESTED** [review]
Please apply the README suggestions to all the READMEs.

Some samples will need to be updated, like how ArcadeEditor.tsx had the imports in it, it should be moved to index.tsx for clarity.

**Review by andygup — APPROVED** [review]
👍 LGTM. Before merging, be sure to remove the `draft` label so that the zip files get rebuilt.

**Review by omarkawach — APPROVED** [review]
LGTM! I updated the React sample comments a bit since the imports sometimes live in a different file.

**Pull request merged**